### PR TITLE
PR fixing and adding Packages tests to e2e

### PR DESCRIPTION
**Pull request by jonahjon**
Signed-off-by: jonahjon <jonahjones094@gmail.com>

*Issue #, if available:*

*Description of changes:*

Fixed changes from other PR which got bugged https://github.com/aws/eks-anywhere/pull/3714.

### CODE REVIEW

File: test/e2e/curated_packages_test.go [hunk 1]
```diff
@@ -187,17 +190,19 @@ func runCuratedPackageInstallSimpleFlow(test *framework.ClusterE2ETest) {
 // decisions about test packages or methodologies, right now.
 
 func TestCPackagesDockerUbuntuKubernetes120SimpleFlow(t *testing.T) {
+	framework.CheckCuratedPackagesCredentials(t)
 	test := framework.NewClusterE2ETest(t,
 		framework.NewDocker(t),
 		framework.WithClusterFiller(api.WithKubernetesVersion(v1alpha1.Kube120)),
-		framework.WithPackageConfig(t, packageBundleURI(v1alpha1.Kube120),
+		framework.WithPackageConfig(t, packageBundleURI(v1alpha1.Kube121),
```
Comment (cxbrowne1207) [inline test/e2e/curated_packages_test.go:197]: Is this correct? Why is this changed to Kube121 when the test is for Kube120?

File: test/e2e/curated_packages_test.go [hunk 2]
```diff
@@ -209,17 +214,19 @@ func TestCPackagesDockerUbuntuKubernetes121SimpleFlow(t *testing.T) {
 }
 
 func TestCPackagesDockerUbuntuKubernetes122SimpleFlow(t *testing.T) {
+	framework.CheckCuratedPackagesCredentials(t)
 	test := framework.NewClusterE2ETest(t,
 		framework.NewDocker(t),
 		framework.WithClusterFiller(api.WithKubernetesVersion(v1alpha1.Kube122)),
-		framework.WithPackageConfig(t, packageBundleURI(v1alpha1.Kube122),
+		framework.WithPackageConfig(t, packageBundleURI(v1alpha1.Kube121),
```
Comment (cxbrowne1207) [inline test/e2e/curated_packages_test.go:221]: Similar to above. Why was this changed from Kube122 to Kube121 when the test is for Kube122?

Comment (jonahjon) [reply test/e2e/curated_packages_test.go:221]: copy paste mistakes, thanks for catching

Comment (jonahjon) [reply test/e2e/curated_packages_test.go]: fixed

File: test/e2e/curated_packages_test.go (hunk 1 above)
Comment (jonahjon) [reply test/e2e/curated_packages_test.go]: fixed

File: cmd/integration_test/build/buildspecs/conformance-eks-a-cli.yml [hunk 3]
```diff
@@ -13,6 +13,8 @@ env:
   secrets-manager:
     EKSA_VSPHERE_USERNAME: "vsphere_ci_beta_connection:vsphere_username"
     EKSA_VSPHERE_PASSWORD: "vsphere_ci_beta_connection:vsphere_password"
+    EKSA_AWS_ACCESS_KEY_ID: "packages_ci_beta:aws_access_key_id"
+    EKSA_AWS_SECRET_ACCESS_KEY: "packages_ci_beta:aws_secret_access_key_id"
```
[the remark this answers is not in the document]
Comment (jonahjon) [reply cmd/integration_test/build/buildspecs/conformance-eks-a-cli.yml:17]: fixed

File: test/framework/curatedpackages.go [hunk 4]
```diff
@@ -25,3 +26,17 @@ func WithPackageConfig(t *testing.T, bundleURI, chartName, chartURI,
 		}
 	}
 }
+
+// CheckCuratedPackagesCredentials will exit out if the Curated Packages environment variables are not set.
+func CheckCuratedPackagesCredentials(t *testing.T) {
+	requiredEnvVars := []string{
+		"EKSA_AWS_SECRET_ACCESS_KEY",
+		"EKSA_AWS_ACCESS_KEY_ID",
+	}
```
[the remark this answers is not in the document]
Comment (jonahjon) [reply test/framework/curatedpackages.go:36]: fixed

File: test/e2e/README.md [hunk 5]
```diff
@@ -51,6 +51,8 @@ VSPHERE_PASSWORD
 T_VSPHERE_CIDR
 GOVC_URL
 T_VSPHERE_CLUSTER_IP_POOL # comma-separated list of CP ip addresses
+EKSA_AWS_ACCESS_KEY_ID # For Packages tests
+EKSA_AWS_SECRET_ACCESS_KEY # For Packages tests
```
[the remark this answers is not in the document]
Comment (jonahjon) [reply test/e2e/README.md:55]: fixed

File: cmd/integration_test/build/buildspecs/conformance-eks-a-cli.yml [hunk 6]
```diff
@@ -13,6 +13,8 @@ env:
   secrets-manager:
```
[the remark this answers is not in the document]
Comment (jonahjon) [reply cmd/integration_test/build/buildspecs/conformance-eks-a-cli.yml:13]: fixed

File: cmd/integration_test/build/buildspecs/conformance-eks-a-cli.yml [hunk 7]
```diff
@@ -13,6 +13,9 @@ env:
...
     EKSA_VSPHERE_USERNAME: "vsphere_ci_beta_connection:vsphere_username"
     EKSA_VSPHERE_PASSWORD: "vsphere_ci_beta_connection:vsphere_password"
+    EKSA_AWS_ACCESS_KEY_ID: "packages_ci_beta:aws_access_key_id"
+    EKSA_AWS_SECRET_ACCESS_KEY: "packages_ci_beta:aws_secret_access_key_id"
+    EKSA_AWS_REGION: "us-west-2"
```
Comment (junshun) [inline cmd/integration_test/build/buildspecs/conformance-eks-a-cli.yml:18]: I believe this should be, 
```suggestion
    EKSA_AWS_REGION: "packages_ci_beta:aws_region"
```
If the intent is not to get this from secret manager and just hardcode, then it needs to be moved up to the variables section instead of secret manager section.

File: cmd/integration_test/build/buildspecs/test-eks-a-cli.yml [hunk 8]
```diff
@@ -35,6 +35,9 @@ env:
   secrets-manager:
     EKSA_VSPHERE_USERNAME: "vsphere_ci_beta_connection:vsphere_username"
     EKSA_VSPHERE_PASSWORD: "vsphere_ci_beta_connection:vsphere_password"
+    EKSA_AWS_ACCESS_KEY_ID: "packages_ci_beta:aws_access_key_id"
+    EKSA_AWS_SECRET_ACCESS_KEY: "packages_ci_beta:aws_secret_access_key_id"
+    EKSA_AWS_REGION: "us-west-2"
```
Comment (junshun) [inline cmd/integration_test/build/buildspecs/test-eks-a-cli.yml:40]: Same here
```suggestion
    EKSA_AWS_REGION: "packages_ci_beta:aws_region"
```

**Review by junshun — CHANGES_REQUESTED** [review]

File: cmd/integration_test/build/buildspecs/conformance-eks-a-cli.yml (hunk 7 above)
Comment (jonahjon) [reply cmd/integration_test/build/buildspecs/conformance-eks-a-cli.yml:18]: ok will change and add that to secret manager

File: cmd/integration_test/build/buildspecs/test-eks-a-cli.yml (hunk 8 above)
Comment (jonahjon) [reply cmd/integration_test/build/buildspecs/test-eks-a-cli.yml]: fixed

**Review by junshun — APPROVED** [review]
